### PR TITLE
Feature: alarming response

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Draft for Feature: alarming response.

### CODE REVIEW

File: Wire-iOS/Sources/Components/Buttons/SpinnerButton.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// A button with spinner at the trailing side. Title text is non trancated.
```
Comment (David-Henner) [inline Wire-iOS/Sources/Components/Buttons/SpinnerButton.swift]:
```suggestion
/// A button with spinner at the trailing side. Title text is non truncated.
```

File: Wire-iOS/Sources/Components/Buttons/SpinnerButton.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    override init() {
+        super.init()
+
+        configureTitleLabel()
+    }
+
+    /// multi line support of titleLabel
+    private func configureTitleLabel() {
+        if let titleLabel = titleLabel {
```
Comment (David-Henner) [inline Wire-iOS/Sources/Components/Buttons/SpinnerButton.swift]: maybe we could use a `guard` here

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Poll/ConversationButtonMessageCell.swift [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+final class ConversationButtonMessageCellDescription: ConversationMessageCellDescription {
+    typealias View = ConversationButtonMessageCell
+
+    var topMargin: Float = Float.ConversationButtonMessageCell.verticalInset
+
+    var isFullWidth: Bool = false ///TODO:
+
+    var supportsActions: Bool = false ///TODO:
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Poll/ConversationButtonMessageCell.swift]: what are the `TODO` for ?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/ConversationMessageCell.swift [hunk 4]
```diff
@@ -200,19 +200,37 @@ extension ConversationMessageCellDescription {
...
+    /// - Returns: true if both self and other having same type
     func isConfigurationEqual(with other: Any) -> Bool {
         return type(of: self) == type(of: other)
     }
     
 }
 
+extension ConversationMessageCellDescription where View.Configuration : Equatable {
+    
+    /// Default impletmentation of isConfigurationEqual
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/ConversationMessageCell.swift]:
```suggestion
    /// Default implementation of isConfigurationEqual
```

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/ConversationMessageCell.swift [hunk 5]
```diff
@@ -200,19 +200,37 @@ extension ConversationMessageCellDescription {
...
-        adapterCell.cellView.configure(with: self.configuration, animated: animated)
+        adapterCell.cellView.configure(with: configuration, animated: animated)
         
         if cell.isVisible {
             _ = message?.startSelfDestructionIfNeeded()
         }
     }
     
+    
+    /// Default impletmentation of isConfigurationEqual. If the configure is Equatable, see below Conditionally Conforming for View.Configuration : Equatable
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/ConversationMessageCell.swift]:
```suggestion
    /// Default implementation of isConfigurationEqual. If the configure is Equatable, see below Conditionally Conforming for View.Configuration : Equatable
```

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Poll/ConversationButtonMessageCell.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Poll/ConversationButtonMessageCell.swift]: They are not needed anymore, thanks

File: Cartfile.resolved [hunk 6]
```diff
@@ -16,10 +16,10 @@ github "wireapp/swift-protobuf" "1.5.0"
...
 github "wireapp/wire-ios-canvas" "10.0.0"
 github "wireapp/wire-ios-cryptobox" "21.1.0"
-github "wireapp/wire-ios-data-model" "215.0.0"
+github "wireapp/wire-ios-data-model" "216.0.0"
 github "wireapp/wire-ios-images" "30.0.0"
 github "wireapp/wire-ios-link-preview" "26.0.0"
-github "wireapp/wire-ios-protos" "21.0.0"
+github "wireapp/wire-ios-protos" "22.0.0"
 github "wireapp/wire-ios-request-strategy" "191.0.0"
 github "wireapp/wire-ios-share-engine" "174.0.0"
```
Comment (typfel) [inline Cartfile.resolved]: Since you release a major `wire-ios-data-model` I would expect a major bump of

wire-ios-share-engine
wire-ios-request-strategy

**Pull request merged**